pysheaf.py

```diff
@@ -52,11 +52,49 @@ def isCoface(self,index,orientation=None):
             return (index,orientation) in [(cf.index,cf.orientation) for cf in self.cofaces]
         
 class CellComplex:
-    def __init__(self,cells=[]):
+    def __init__(self,cells=None):
         """Construct a cell complex from its cells"""
-        self.cells=cells
+        if cells == None:
+            self.cells=[]
+        else:
+            self.cells=cells
+
+        # Register cell names with indices into the self.cells list        
+        self.cell_dict={}
+        self.coface_dict={}
+        for i,c in enumerate(self.cells):
+            if c.name == None: # Build names if not present
+                c.name = str(i)
+            self.cell_dict[c.name]=i
+
+    def add_cell(self,cell):
+        """Add a cell to the cell complex"""
+
+        if cell.name == None: # Construct a name if needed
+            cell.name = str(len(self.cells))
+
+        self.cell_dict[cell.name]=len(self.cells)
+        self.cells.append(cell)
+
+    def add_cells_from(self,cells):
+        for c in cells:
+            self.add_cell(c)
+
+    def add_coface(self,names,coface):
+        """Add a coface to the cell complex, referenced by a pair of cell names.  The names argument is assumed to be a pair: (face,coface).  The .index attribute for the coface is ignored.  If the cells aren't present, this will raise KeyError."""
+        # Look up which cells are involved...
```

Comment:
You need UID's to reference a cell. In theory the cells will all have unique names but there is no check for this in the Cell constructor or CellComplex constructor. If you are just using ID's (or even positions) as names this isn't an issue. It might be better in general to use the ID as the key to reference a cell.